ghost/core/core/server/services/comments/CommentsController.js

```diff
@@ -35,6 +35,22 @@ module.exports = class CommentsController {
      * @param {Frame} frame
      */
     async browse(frame) {
+        if (frame.options.post_id) {
+            if (frame.options.filter) {
+                frame.options.mongoTransformer = function (query) {
+                    return {
+                        $and: [
+                            {
+                                post_id: frame.options.post_id
+                            },
+                            query
+                        ]
+                    };
+                };
```

Comment:
Is it worth adding a TODO comment here to clean this up when the old route is removed? Also wondering if we should do anything if the supplied filter already has a `post_id` in it 🤔

Reply:
Reading again I'm not sure this comment makes sense as I can't see how this would really change when the old route is removed 😄

Still wonder if it's worth adding some protection against double `post_id=x AND post_id=y` in the filter but that can be a cleanup thing.

Reply:
I think we'll still need this even when the old route is removed! This is the mechanism that we limit the browse to the requested post

Yeah so, what do you think we should do? We could either error, or strip out any post_id filter operations.

I'm deffo in favour of erroring when invalid filters are used, BUT it goes against Postels law of liberal with what we accept...


I can fairly easily strip out any post_id filter operations if we think that's valuable

Reply:
As discussed in slack - we're gonna leave this for now to keep changeset small